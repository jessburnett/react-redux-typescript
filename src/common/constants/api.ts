const apiBaseUrl = '/api';
export const dataApiBaseUrl = `${apiBaseUrl}/get-data`;
