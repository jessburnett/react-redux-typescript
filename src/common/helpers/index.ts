import { testGeneratorStep } from './tests';

export {
  testGeneratorStep,
};
