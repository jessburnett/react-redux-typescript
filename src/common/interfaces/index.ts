import { IAction, IEmptyAction, IFailureAction } from './IAction';
import { IError } from './IError';
import { IImmutableArray } from './IImmutableArray';
import { IImmutableObject } from './IImmutableObject';

export {
  IAction,
  IEmptyAction,
  IFailureAction,
  IError,
  IImmutableArray,
  IImmutableObject,
};
