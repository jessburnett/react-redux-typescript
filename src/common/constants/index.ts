import * as actions from './actions';
import { dataApiBaseUrl } from './api';

export {
  actions,
  dataApiBaseUrl,
};
