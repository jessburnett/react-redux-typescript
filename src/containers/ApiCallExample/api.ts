import axios from 'axios';

import { IDataExample } from './index';

import { dataApiBaseUrl } from 'common/constants';

const parseData = (data: IDataExample): IDataExample => data.filter((item) => item % 2 === 0);

export const fetchData = (): Promise<IDataExample> => axios.get(dataApiBaseUrl).then((response) => parseData(response.data));
