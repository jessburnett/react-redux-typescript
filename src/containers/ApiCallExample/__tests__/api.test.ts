import axios from 'axios';
import axiosMockAdapter from 'axios-mock-adapter';

import { dataApiBaseUrl } from 'common/constants';

const testData = [1,2,3,4,5];
const axiosMock = new axiosMockAdapter(axios);
axiosMock.onGet(dataApiBaseUrl).reply(200, testData);

import { fetchData } from '../api';

describe('ApiCallExample API', () => {

  it('calls and handles fetchData result correctly', () => {
    return fetchData().then((result) => {
      expect(result).toMatchSnapshot();
    });
  });

});
