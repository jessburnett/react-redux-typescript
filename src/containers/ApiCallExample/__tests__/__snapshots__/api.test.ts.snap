exports[`ApiCallExample API calls and handles fetchData result correctly 1`] = `
Array [
  2,
  4,
]
`;
